core/schema/module.go

```diff
@@ -282,6 +292,7 @@ func (s *moduleSchema) typeDefWithObject(ctx context.Context, def *core.TypeDef,
 	Name        string
 	Description string `default:""`
 	SourceMap   dagql.Optional[core.SourceMapID]
+	Deprecated  string `default:""`
```

Comment:
Should be optional too

Reply:
In order to do that, I also had to change the dagql representation of the DeprecatedReason. (see associated commit, starting with `dagql(*)`. It is actually more compliant with the graphQL spec